src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1460 @@
+        self.assertEqual(self.MirrorDescanner.scanGain.value, (0.9 * max_scanGain, 0.9 * max_scanGain))
+
+        # Check if VA refuses to set limits outside allowed range
+        with self.assertRaises(IndexError):
+            self.MirrorDescanner.scanGain.value = (1.2 * max_scanGain, 1.2 * max_scanGain)
+        self.assertEqual(self.MirrorDescanner.scanGain.value, (0.9 * max_scanGain, 0.9 * max_scanGain))
+
+        with self.assertRaises(IndexError):
+            self.MirrorDescanner.scanGain.value = (1.2 * min_scanGain, 1.2 * min_scanGain)
+        self.assertEqual(self.MirrorDescanner.scanGain.value, (0.9 * max_scanGain, 0.9 * max_scanGain))
+
+    def test_getXAcqSetpoints(self):
+        """
+        After each change the same properties of the testcases are checked, first the length (total number of setpoints)
+        then the expected range of the values are checked. The maximum is not necessarily reached because this is
+        dependent on the descan_period/dwell_time
+        """
+        descanner = self.MirrorDescanner
+        scanner = self.EBeamScanner
+        mppc = self.MPPC
+        # Change values such that it is easy to follow the calculation
+        descanner.scanOffset.value = (0.09767299916075389, 0.09767299916075389)
+        descanner.scanGain.value = (0.646387426565957, 0.646387426565957)
+        scanner.dwellTime.value = descanner.clockPeriod.value
+        def expected_setpoint_length(dwellTime, physcicalFlybackTime, X_cell_size, descan_period):
+            # Ceil round the number of scanning points so that if a half descan period is left at least a full extra
+            # setpoint is added to allow the scan to be properly finished
+            scanning_setpoints = math.ceil(numpy.round((dwellTime * X_cell_size) / descan_period, 10))
+            flyback_setpoints = math.ceil(physcicalFlybackTime / descan_period)
+            return scanning_setpoints + flyback_setpoints
+
+        # Check default values
+        X_descan_setpoints = descanner.getXAcqSetpoints()
+        self.assertEqual(len(X_descan_setpoints),
+                         expected_setpoint_length(scanner.dwellTime.value,
+                                                  descanner.physicalFlybackTime,
+                                                  mppc.cellCompleteResolution.value[0],
+                                                  descanner.clockPeriod.value))
+
+        self.assertEqual(min(X_descan_setpoints),
+                         numpy.floor(convertBits(descanner.scanOffset.value,
+                                                  numpy.array(descanner.scanOffset.range)[:, 1])[0]))
+        self.assertEqual(max(X_descan_setpoints),
+                         numpy.floor(convertBits(descanner.scanGain.value,
+                                                  numpy.array(descanner.scanGain.range)[:, 1])[0]))
+
+        # Check with changing the dwell_time
```

Comment:
Maybe add, that you do the extensive testing in the hope to catch also floating point errors.